include/veriblock/blockchain/pop/fork_resolution.hpp

```diff
@@ -462,7 +473,7 @@ struct PopAwareForkResolutionComparator {
 
           return true;
         },
-        [&] { removePayloads(index, payloads); });
+        [&] { removePayloads(index, payloads, lastApplied); });
```

Comment:
More logical solution is to maintain a local vector of added payloads and pass it to the removePayloads instead of using additional lastApplied index.